src/dmd/func.d

```diff
@@ -104,7 +104,8 @@ public:
     override void visit(TryFinallyStatement s)
     {
         DtorExpStatement des;
-        if (fd.nrvo_can && s.finalbody && (des = s.finalbody.isDtorExpStatement()) !is null && fd.nrvo_var == des.var)
+        if (fd.nrvo_can && s.finalbody && (des = s.finalbody.isDtorExpStatement()) !is null &&
+            fd.nrvo_var == des.var && !global.params.betterC)
```

Comment:
Can betterC not be used as a feature gate here?

Reply:
Would `&& global.params.useExceptions` be appropriate?

Reply:
done